src/Sentry.Unity.Editor.iOS/SentryXcodeProject.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.IO;
+using UnityEditor.iOS.Xcode;
+using UnityEditor.iOS.Xcode.Extensions;
+using UnityEngine;
+
+namespace Sentry.Unity.Editor.iOS
+{
+    internal class SentryXcodeProject : IDisposable
+    {
+        private const string FrameworkName = "Sentry.framework";
+
+        private const string MainPath = "MainApp/main.mm";
+        private const string OptionsPath = "MainApp/SentryOptions.m";
+        private string UnityPackageFrameworkRoot = Path.Combine("Frameworks", "io.sentry.unity");
+
+        private readonly string _projectRoot;
+        private readonly SentryUnityOptions _options;
+
+        private readonly PBXProject _project;
+        private readonly string _projectPath;
+
+        internal string? RelativeFrameworkPath { get; set; }
+
+        private readonly INativeMain _nativeMain;
+        private readonly INativeOptions _nativeOptions;
+
+        internal SentryXcodeProject(
```

Comment:
Communicates intent: This ctor is designed to be used by those with access to this class:

```suggestion
        public SentryXcodeProject(
```

If the class became public, nothing else needs to change.

And the other `internal` ctor is clear to be there for 'internal' (i.e: Tests or what not) reasons